File: lession6/Chain Of Responsibility/java/Command/src/main/java/SubstractCommand.java

public class SubstractCommand implements Command{

    private OperationApi operation = null;

    private int opeNum;

    public SubstractCommand(OperationApi operation,int opeNum){
        this.operation = operation;
        this.opeNum = opeNum;
    }

    public void execute() {
        this.operation.substract(opeNum);
    }

    public void undo() {
        this.operation.add(opeNum);
    }
}

File: lession6/Memento/csharp/Caculator/Caculator/SubstractCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


public class SubstractCommand : AbstractCommand
{
    private int opeNum;
    public SubstractCommand(OperationApi operation, int opeNum)
        : base(operation)
    {
        this.opeNum = opeNum;
    }
    public override void Execute()
    {
        this.operation.Substract(this.opeNum);
    }
}

